lib/iris/experimental/ugrid/mesh.py

```diff
@@ -2784,6 +2784,11 @@ def fix_repr(val):
                     )
                     raise ValueError(msg)
 
+        # Don't use 'coord_system' as a constructor arg, since for
+        # MeshCoords it is deduced from the mesh.
+        # (Otherwise a non-None coord_system breaks the 'copy' operation)
+        use_metadict.pop("coord_system")
```

Comment:
Does this fix a bug which already existed? If so, it might be worth adding a bugfix part to the whatsnew.

Reply:
Also, is there a test which shows that this copy operation is working correctly?

Reply:
> Does this fix a bug which already existed? If so, it might be worth adding a bugfix part to the whatsnew.

Well, effectively #5557 is that bug.
Since you can't get a mesh coordinate with a coord-system from loading a file, manual assigning one is the only way you would hit this problem, at present.

Reply:
> is there a test which shows that this copy operation is working correctly

I guess not. I will add one.